### PR TITLE
drivers/sdcard_spi: use of ucrc16 by default

### PR DESCRIPTION
**Pull request by cyrilfr**
### Contribution description

We probably should use checksum algorithms provided in RIOT by default. Here the standard lightweight CCITT CRC16 is used to avoid code duplication.

The CRC7 isn't implemented yet. Maybe a candidate for a future refactoring?

### CODE REVIEW

**Comment by MichelRottleuthner** [comment]
Thank you for addressing this! I had that in mind for a while now, but never took the time to actually do it - shame on me! :). 
>The CRC7 isn't implemented yet. Maybe a candidate for a future refactoring?

Yes! there are also a few more things in this driver that could be refactored. For reference the change from char to uint8_t discussed in #7853, some improvements regarding timeout handling, performance tweaks and while at it - some code cleanup. But one thing at a time...

File: drivers/sdcard_spi/sdcard_spi.c [hunk 1]
```diff
@@ -622,9 +606,9 @@ static sd_rw_response_t _read_data_packet(sdcard_spi_t *card, char token, char *
 
         char crc_bytes[2];
         if (_transfer_bytes(card, 0, crc_bytes, sizeof(crc_bytes)) == sizeof(crc_bytes)) {
-            uint16_t data__crc_16 = (crc_bytes[0] << 8) | crc_bytes[1];
+            uint16_t data_crc16 = (crc_bytes[0] << 8) | crc_bytes[1];
 
-            if (_crc_16(data, size) == data__crc_16) {
+            if (ucrc16_calc_be((const uint8_t *)data, size, UCRC16_CCITT_POLY_BE, 0) == data_crc16) {
```
Comment (MichelRottleuthner) [inline drivers/sdcard_spi/sdcard_spi.c]: `const` isn't really required here, right?

File: drivers/sdcard_spi/sdcard_spi.c [hunk 2]
```diff
@@ -712,8 +696,8 @@ static sd_rw_response_t _write_data_packet(sdcard_spi_t *card, char token, const
 
     if (_transfer_bytes(card, data, 0, size) == size) {
 
-        uint16_t data__crc_16 = _crc_16(data, size);
-        char crc[sizeof(uint16_t)] = { data__crc_16 >> 8, data__crc_16 & 0xFF };
+        uint16_t data_crc16 = ucrc16_calc_be((const uint8_t *)data, size, UCRC16_CCITT_POLY_BE, 0);
```
Comment (MichelRottleuthner) [inline drivers/sdcard_spi/sdcard_spi.c]: same as above. It does no harm, but for readability I'd remove the 'const' here.

**Comment by MichelRottleuthner** [comment]
looks like something went wrong while squashing. Did you follow the procedure described [here](https://github.com/RIOT-OS/RIOT/wiki/Git-cheatsheet)?

**Comment by cyrilfr** [comment]
Yes, but it does not seem to work...

**Comment by MichelRottleuthner** [comment]
Congrats to your first contribution to RIOT :tada:

**Pull request merged**